### ISSUE TITLE
FP S4226: "move the method inside the class itself" only possible if class is defined in source

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#6430]
### Description

The rule description for S4226 says: 

> It makes little sense to create an extension method when it is possible to just add that method to the class itself. This rule raises an issue when an extension is declared in the same namespace as the class it is extending.

This only works if the extended class is available as source code.

### Repro steps
Consider the following project structure:

Assembly1 (e.g., Nuget) or generated code:
```cs
namespace A;

class Person
{
  // ...
}
```
Assembly2:
```cs
namespace A;
public static class PersonExtensions
{
    public static void Promote(this Person, string newPosition) { } // FP: S4226  is raised here
}
```

### Expected behavior

Don't raise if [ISymbol.DeclaringSyntaxReferences](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.isymbol.declaringsyntaxreferences?view=roslyn-dotnet-4.3.0) for the extended type is empty or points to generated code exclusively.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1801](https://sonarsource.atlassian.net/browse/NET-1801)

[NET-1801]: https://sonarsource.atlassian.net/browse/NET-1801?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
